src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1318,7 +1318,6 @@ void TorrentHandle::resume_impl(bool forced, bool uploadMode)
     }
 
     m_nativeHandle.auto_managed(!forced);
-    m_nativeHandle.set_upload_mode(uploadMode);
```

Comment:
It was necessary to prevent an unexpected data download in the interval between the recheck end and re-pause.

Reply:
>It was necessary to prevent an unexpected data download in the interval between the recheck end and re-pause.

Just to be sure: We have another mechanism to prevent his corner-case?

Reply:
>We have another mechanism to prevent this corner-case?

There is no such corner-case now since it is paused by libtorrent itself in its io thread. Previously, we had a gap between the end of the recheck and the processing of the alert (and one more gap between calling pause() in app thread and pausing it in io thread).